stream.go

```diff
@@ -102,26 +105,36 @@ func NewClientStream(ctx context.Context, desc *StreamDesc, cc *ClientConn, meth
 	if err != nil {
 		return nil, toRPCErr(err)
 	}
+	if EnableTracing {
+		trInfo.tr = trace.New("Sent."+methodFamily(desc.StreamName), desc.StreamName)
+		trInfo.firstLine.client = true
+		if deadline, ok := ctx.Deadline(); ok {
+			trInfo.firstLine.deadline = deadline.Sub(time.Now())
+		}
+		trInfo.tr.LazyLog(&trInfo.firstLine, false)
+	}
 	s, err := t.NewStream(ctx, callHdr)
 	if err != nil {
 		return nil, toRPCErr(err)
 	}
 	return &clientStream{
-		t:     t,
-		s:     s,
-		p:     &parser{s: s},
-		desc:  desc,
-		codec: cc.dopts.codec,
+		t:         t,
+		s:         s,
+		p:         &parser{s: s},
+		desc:      desc,
+		codec:     cc.dopts.codec,
+		traceInfo: trInfo,
```

Comment:
this is incorrect. This is making a copy of the traceInfo struct, but the trace has a pointer to the original, so any changes to firstLine after this will not be reflected, for instance. It'll also pin both bits of memory.

Either clientStream needs to be allocated higher up (probably the simplest option), or it needs to have a *traceInfo field instead.